### PR TITLE
templater: add "parents" keyword in place of "parent_commit_ids"

### PR DESCRIPTION
**Pull request by yuja**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: tests/test_commit_template.rs [hunk 1]
```diff
@@ -40,16 +40,39 @@ fn test_log_parent_commit_ids() {
...
+
+    // Commit object isn't printable
+    let stderr = test_env.jj_cmd_failure(&repo_path, &["log", "-T", "parents"]);
+    insta::assert_snapshot!(stderr, @r###"
+    Error: Failed to parse template:  --> 1:1
+      |
+    1 | parents
+      | ^-----^
+      |
+      = Expected expression of type "Template"
```
[the remark this answers is not in the document]
Comment (necauqua) [reply tests/test_commit_template.rs:62]: If it's impossible to do then current message is fine.

I see, however that your `try_` methods return `Option`s, if they returned `Result<_, TemplateParseError>` or something similar throughout and the error had an optional hint field, then [here](https://github.com/martinvonz/jj/pull/1414/files#diff-86f613aa922caaa543310be4ff32cf3a93ee434a2cea6a1dce4619073a008f0fR157-R158) you'd be able to set it to `maybe you want .commit_id()` and `maybe you want .map(..)`

Just a suggestion without much context, I'm not acquainted with the codebase at all ¯\\\_(ツ)_/¯

[the remark this answers is not in the document]
Comment (yuja) [reply tests/test_commit_template.rs:62]: I know, but `TemplateParseError` can't be instantiated without span atm, so some more refactoring needed. It's also not good to suggest `.commit_id()`, and we'll probably need to show (say, up to 5) method candidates.

[the remark this answers is not in the document]
Comment (martinvonz) [reply tests/test_commit_template.rs:62]: I'm guessing #1418 is related (to prepare for printing similar hints for template errors). IMO, feel free to merge this as is. It sounds like @necauqua agrees.

**Review by martinvonz — APPROVED** [review]

File: tests/test_commit_template.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (yuja) [reply tests/test_commit_template.rs:62]: Yeah, #1418 is the easy one. I'll revisit template error hints later.

**Pull request merged**